gamemodes/terrortown/gamemode/roles/zombie/zombie.lua

```diff
@@ -364,7 +364,7 @@ hook.Add("DoPlayerDeath", "Zombie_DoPlayerDeath", function(victim, attacker, dmg
     if not IsValid(inflictor) or WEPS.GetClass(inflictor) ~= "weapon_zom_claws" then return end
 
     -- If they were killed by a normal slash attack or spit conversion is enabled and they were killed by spitting, convert them
-    if dmginfo:IsDamageType(DMG_SLASH) or (zombie_spit_convert:GetBool() and dmginfo:IsDamageType(DMG_BULLET)) then
+    if not attacker:IsRoleAbilityDisabled() and (dmginfo:IsDamageType(DMG_SLASH) or (zombie_spit_convert:GetBool() and dmginfo:IsDamageType(DMG_BULLET))) then
```

Comment:
Move the more expensive check later
```suggestion
    if (dmginfo:IsDamageType(DMG_SLASH) or (zombie_spit_convert:GetBool() and dmginfo:IsDamageType(DMG_BULLET))) and not attacker:IsRoleAbilityDisabled() then
```